src/rt/lifetime.d

```diff
@@ -1193,24 +1186,53 @@ extern (C) void _d_delarray_t(void[]* p, const TypeInfo_Struct ti)
 {
     if (p)
     {
-        assert(!(*p).length || (*p).ptr);
-        if ((*p).ptr)
+        auto bic = __getBlkInfo(p.ptr);
+        auto info = bic ? *bic : GC.query(p.ptr);
+
+        if (info.base && (info.attr & BlkAttr.APPENDABLE))
```

Comment:
It could be used, but it would not allow to manually emplace structs into a byte array and delete them later:
```
auto foo = new ubyte[2*Bar.sizeof];
auto bars = cast(Bar[])foo;
delete bars;
```